### PR TITLE
Factorize the construction and definition of JDBC urls

### PR DESCRIPTION
**Pull request by murdos**
We have many templates that uses a JDBC url depending on the database type.
This PR creates a function dedicated to that responsibility, used by the templates.

Because of this duplication, I think, there was subtle differences in options used for MySQL and MariaDB URL, I chose to uniformize them, but your feedback is welcome.

-   [x] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
-   [x] Tests are added where necessary
-   [x] Documentation is added/updated where necessary
-   [x] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/master/CONTRIBUTING.md) are followed

### CODE REVIEW

File: generators/server/templates/src/main/resources/config/application-dev.yml.ejs [hunk 1]
```diff
@@ -105,31 +105,31 @@ spring:
...
         username: root
         password:
         <%_ } else if (devDatabaseType === 'postgresql') { _%>
-        url: jdbc:postgresql://localhost:5432/<%= baseName %>
+        url: <%- getJDBCUrl(devDatabaseType, { databaseName: baseName, hostname: 'localhost' }) %>
         username: <%= baseName %>
         password:
         <%_ } else if (devDatabaseType === 'oracle') { _%>
-        url: jdbc:oracle:thin:@localhost:1521:xe
+        url: <%- getJDBCUrl(devDatabaseType, { databaseName: baseName, hostname: 'xe' }) %>
```
Comment (mshima) [inline generators/server/templates/src/main/resources/config/application-dev.yml.ejs:120]:
```suggestion
        url: <%- getJDBCUrl(devDatabaseType, { databaseName: 'xe', hostname: 'localhost' }) %>
```

Comment (murdos) [reply generators/server/templates/src/main/resources/config/application-dev.yml.ejs]: Thanks @mshima for spotting this error!

**Comment by MathieuAA** [comment]
Can you add some tests for the function?

**Comment by murdos** [comment]
@MathieuAA : done 😉

**Pull request merged**